mountinterface.go

```diff
@@ -88,3 +88,16 @@ type MountResponse struct {
 	FileHandle
 	AuthFlavors []int
 }
+
+// Group contains a group node, with information about the allowed access group.
+type Group struct {
+	Name []byte
+	Next []Group
+}
+
+// Export contains an export node, with information about an exported filesystem.
+type Export struct {
+	Dir     []byte
+	Groups  []Group
```

Comment:
I believe this is `*Group` rather than `[]Group` - in that it is either `nil` or the next group in a linked list, right?

Reply:
It is, but the Go wrapper "helpfully" dereferences all pointers - so we declare it as an array instead.

Reply:
They are documented as equivalent in XDR: https://datatracker.ietf.org/doc/html/rfc1014#section-3.18

Reply:
The pattern in the rest of this library has been to write the individual fields with the extra 0 to signify the end of the list

e.g. https://github.com/willscott/go-nfs/blob/master/nfs_onreaddir.go#L116

while this is more fiddly, it makes the type 'logically correct', so that the caller doesn't have to understand the nuance.